src/instance-lookup.js

```diff
@@ -52,7 +73,23 @@ function instanceLookup(server, instanceName, callback, timeout, retries) {
       socket = dgram.createSocket('udp4');
       socket.on('error', onError);
       socket.on('message', onMessage);
-      socket.send(request, 0, request.length, SQL_SERVER_BROWSER_PORT, server);
+
+      if (multiSubnetFailover) {
```

Comment:
MultiSubnetFailover is handled differently here vs in Connector. In connector, we call lookupAll for both cases and then fire requests in parallel or in sequence depending on whether multiSubnetFailover succeeds or not. Whereas here, we only try one IP address for multiSubnetFailover == false. I think we should keep these consistent.

Also we're doing lookupAll twice for instance-lookup code path. Once in instanceLookup() method and then again in connectOnPort(). We should only need to do this once I think. One way to do that would to invoke lookupAll in connect() method for all cases and then pass addresses as a parameter for connectOnPort() method as well as instance-lookup().

One other suggestion for the instance-lookup case would be to use the same IP address that returned the port, in connectOnPort. I'm not sure if it's mandated but seems safe to do.